src/reactioncoordinate.cpp

```diff
@@ -328,19 +328,21 @@ MoleculeProperty::MoleculeProperty(const json &j, Space &spc) : ReactionCoordina
     else if (property == "Rinner") {
         dir = j.at("dir");
         indexes = j.value("indexes", decltype(indexes)());
-        assert(indexes.size() == 2 && "An array of 2 indexes should be specified.");
-        function = [&spc, &dir = dir, i = indexes[0], j = indexes[1]]() {
+        assert(indexes.size() >= 3 && "An array of at least 3 indexes should be specified.");
+        function = [&spc, &dir = dir, i = indexes[0], j = indexes[1], k = indexes[2], l = indexes[3]]() {
             Average<double> Rj, Ri;
             auto slicei = spc.findAtoms(i);
             auto cm = Geometry::massCenter(slicei.begin(), slicei.end(), spc.geo.getBoundaryFunc());
             auto slicej = spc.findAtoms(j);
             for (auto p : slicej)
```

Comment:
spell out `p` to `const auto &particle`. Also replace `Rjavg` with `const auto mean_radius = Rj.avg`